### DISCUSSION TITLE
Slime: New Feature

### DISCUSSION THREAD

**Discussion by LittleEins** [sunflower-land/sunflower-land#1326] [Ideas]
# Slime - New Feature
![184480555-8e8d363d-dc4c-44c4-aa22-c0934eb0762d](https://user-images.githubusercontent.com/104184348/184502928-6dca7ba8-1069-44dd-b583-6a17c874a0d1.gif)

Slimes are one of the things that completes every fantasy lores. They have many characteristics and utility in every different games. Sometimes they are monsters in gaining EXP, or they are pets. I proposed this feature because I thought that this one can compliment the Potions Feature that is coming.
## Description
![Slme](https://user-images.githubusercontent.com/104184348/184952409-83d766c0-d375-4172-9e88-c6710b33cae6.gif)

Slimes are docile creature in the world of Sunflower Land, it is one of every Bumpkin's first pet where they help in many ways.

This fascinating creature can help balance the supply of crops in the game and also encourages fellow Bumpkin to save for Slime's molted product which is **'Slime Gel'**.
Here is the Mechanics that will then be implemented for this lovely creatures:
## How to mint slimes?
We can use the Mysterious Trader as the NPC that will give **Petri Dish** that **requires 15 SFL and 250 wood(or depending on the Developers).**
### Petri Dish - Description
"Used to create Slimes."
![petridish-400](https://user-images.githubusercontent.com/104184348/185853296-a8aedf39-859c-48a9-a3f9-905edd127a66.png)

- Petri Dish can then be exchanged to the Scientist NPC for a Slime.(Limited to one stock every 7 days//create craftedAt to save the time when it was last crafted , we use rare crafting as a reference for this.)
- Petri Dish is tradeable(will only be activated after 7 days through hard code) but Slimes are not tradeable.
**You can only have 1 Slime in 1 farm only.**

#### Why Petri Dish in Mysterious Trader in the first 7 days??
We want to make this item limited and make sure that slime quantity will be balanced. This would also be an experiment to see if there's still more active players. This active players will be the one who will be rewarded through minting new Petri Dish for those who missed the 7 day offer of Mysterious Trader. 
It also makes sense that Petri Dish should only be withdrawable after 7 days to also see if they are burned the first 7 days for slimes or withdraw it just for a little profit(Selling it would be a mistake).

## What is the utility of Slime in SFL?
The utility of Slimes in SFL is they help burn crops (and also resources in the future) which rewards player a 'slime gel'.
The 'slime gel' is rewarded to the player based on the amount of crops fed to their Slimes.
![Slme-Molt-sample](https://user-images.githubusercontent.com/104184348/185204110-4fcef9e7-7614-4f99-b072-7bf56badc559.gif)

### Slime Gel - Description
![Slime-Gel-Raw-400x](https://user-images.githubusercontent.com/104184348/184936048-a36111d6-4c27-427b-936d-596df67b39a2.png)
Raw File: ![Slime-Gel-Raw](https://user-images.githubusercontent.com/104184348/184936077-3014cf04-98b5-4b7b-bb92-841cfec886f6.png)
**"An all purpose gel"**
So how do we produce this gels via crops? There would be a new feature that will support this which is the **'Feeding Scale/Measure'**
## Feeding Scale
This feature will help fellow Bumpkins to strategize on what crops they would feed to their lovely Slimes.
The purpose of this Scale is:
- Prevent abuse of Feeding Slime continuosly ( We don't want animal abuse in our world!)
- Create a strategy on what crops will the Bumpkins would plant and fed to their Slimes
- Balance the supply of crops and also rewards active player who feed their Slimes

Bumpkins will need to feed their Slimes with crops and each crop has a points allocated that should fill the meter.
![image](https://user-images.githubusercontent.com/104184348/185303979-61f49819-92d9-4874-9b1e-b93c87259468.png)
Depending on the meter filled via this points allocated is the randomized slime gel production.
### Feeding UI
A **maximum of 3 types of crops can only be used to allocate points**. Which means **the UI will only have 3 slots** that players can use to feed their Slimes. This modal/UI will only show when the Slime is clicked and not fed. We can use the hold interaction to allocate crops in the slots like in the Withdrawing of Resources.
A confirm/cancel button will then show up after choosing the crops that will be fed on the Slime. 
The meter will also have a logic where the more points allocated, the more the output or rewards.
### Points Range:
![image](https://user-images.githubusercontent.com/104184348/185305885-2d446a00-fea4-41ab-8e38-ce962460d36a.png)
Meaning:
- 1-5.99 points can generate 0.125-0.25 slime gel
- 6-12.99 points can generate 0.25-0.5 slime gel
- 13-25.99 points can generate 0.5-1 slime gel
- 26-35.99 points can generate 1-1.5 slime gel
- 36-50.99 points can generate 1.5-2 slime gel
- 51-65.99 points can generate 2-2.5 slime gel
- 66-75.99 points can generate 2.5-3 slime gel
- 76-85.99 points can generate 3-3.5 slime gel
- 86-100 points can generate 3.5-4 slime gel

### Machine Logic:
In this scale, the closer the points is in the maximum value of a pointed range the higher the chance to get the maximum output of slime gel.(Note for coders, I think the best thing to get the range is by switch case? or its up to you xD)
A = Points Allocated
B = Maximum Value of Range 
Formula:
(A / B) * 100 = C or Percentage of getting the Maximum Output 
After getting this numbers we can have this:
A1 = Max Output(Slime Gel) for Range??
A2 = Min Output(Slime Gel) for Range??
D = C
We will use E = Random Number Generator from 1-100:
- IF(E <= D) Give Maximum Output(A1), 
- ELSE Give Minimum Output(A2)

In this way, fellow Bumpkins won't just allocate an exploitable point allocations in the scaling system. This is also gives the idea of Bumpkins in calculating their risks in producing this lovely product.

#### **Feeding Time**
To prevent over minting of slime gel, there will be **no time scaling** which means player will think what food points will they offer to their lovely slimes.
#### **Limitations**
After feeding , Slimes will molt their body **after 6 hours** which will be the Slime Gel.
To prevent the abuse of this new feature, **Slime can have a upset stomach if fed with ranges of more than the maximum seed supply in shop**. 
![image](https://user-images.githubusercontent.com/104184348/185302569-c830cdfd-1f0f-431d-897e-8b699fa1eb53.png)
In other words, you can't feed your Slime with more than 400 sunflower, 200 potato, 100 pumpkin, etc. This is to provide a fair start for both old and new players and also requires luck/randomness in the reward.
#### **What would happen if I ignore this limitations?**
If you fed your slime over this threshold, a debuff will most likely trigger to give consequences to this abusive Bumpkin owners!
This table will show the debuffs and trigger of this limitations:
![image](https://user-images.githubusercontent.com/104184348/185757869-bdeb8294-efdb-48eb-9ffe-c58024ed8226.png)
#### So how do we know that we are already above the threshold? 
We will simply use this Machine Logic for Threshold:
Let X = number of slots occupied(max of 3).//IF slot 1 has plant.name.size = 1 , X=X+1;
1. IF slots X is occupied, get the highest percentage threshold of the X number of crops based on this formula: 
Upset Stomach Threshold Formula:
A = user input size of plant.name
B = plant.name.threshold size
Thus: A / B * 100 = Percentage Threshold
Then, C = Percentage Threshold
//container[x] = C;
We will do this X times and then get the highest percentage between this X containers.
Then, D = highest percentage threshold crops
2. After that, we will then use the tables above to get the level of debuff.
IF D <= 100 -> No debuff! //slime.produce=1.0;slime.duration=1.00;
IF D > 100 -> Reduce Slime Production by 25%.//slime.produce=0.75;slime.duration=1.0;
IF D > 100 -> Reduce Slime Production by 25% and Add 25% more time to produce slime gel..//slime.produce=0.75;slime.duration=1.25;
IF D > 100 -> Reduce Slime Production by 50% and Add 50% more time to produce slime gel.//slime.produce=0.50;slime.duration=1.50;
By employing this debuff logic in slimes, we are ensured that there will be a balance to slime production and no to abusing this bouncy slimes <3 We want to make sure slimes are healthy and the economy should also be.
## So what is the use case of Slime Gel of Slimes?
**Slime gels** is an all purpose resource in the world of Sunflower Land. They are goblin's favorite dessert when cooked in the Kitchen.
Here are some use cases of this all purpose gel.
## Phase 2: Potions 101 - Mechanics & Logics and Petri Dish
- Slime gels can be used to craft Potions through the **Scientist NPC** that will help Bumpkins in their farm productions. In other words, slime gels could be an essential requirements to craft potions! **Ingredients will be up for the developers for this.**
![Scientist-NPC](https://user-images.githubusercontent.com/104184348/185855925-69f2975a-3ada-4a07-b2e9-0a11d4afade6.gif)

### Potions
- Potions is one of the upcoming features of the game soon to help farmers in their farms. With that in mind we also don't want over inflation of this potions to make farmers abuse it to mint more tokens. We want to limit this expected inflation by adding **Failure Rate Mechanics to the production of this Potions**, here is a proposed table for this failure mechanics
#### Randomness Table for Potions
![image](https://user-images.githubusercontent.com/104184348/185981530-ad521d98-b618-4afa-9f79-fa8f2119bf42.png)
The table above shows the possible chances and output of crafting Potions. This provides risk and reward in crafting potions. We want farmers to feel the success,failure and luckiness of this mechanics.
Also, we can craft this potion in the UI / Modal for the Scientist NFT. Potions will only have two functions which is Craft 1 and Craft 5. 
**Craft 1** has a base chances of:
50% SUCCESSFUL-> 3 Potions, 
30% FAILURE -> 2 Potions, and;
20% LUCKY -> 5 Potions
While, **Craft 5** has a base chances of:
50% SUCCESSFUL-> 15 Potions, 
30% FAILURE -> 10 Potions, and;
20% LUCKY -> 20 Potions
In this way, we can balance also the production of slime gels and potions.
#### Petri Dish
Also, **Petri Dish as mentioned earlier can be craftable**, New Slimes can then be minted by crafting Petri Dish in  Scientist NPC which requires: **100 Slime Gels* 
Petri Dish will also have the same mechanics with Potions.
![image](https://user-images.githubusercontent.com/104184348/185984819-86759b5b-0f09-47fd-b645-1e9962bbfaf8.png)
This table above explains that Petri Dishes have only 20% chance to be successfully crafted, 50% to fail but returning 75 Slime Gel and 30% to fail but returning 50 Slime Gel. 
Other infos and machine logics:
-  Petri Dish is used to mint new slimes for new players(or hold to be able to use when more slots of slimes will be available in the farm)
- Only has **1 stock in the Scientist NPC per sync / session**
- If there is 1 Petri Dish in the bag, Crafting button will be disabled.

This means that players who got the Petri Dish from Mysterious Trader will be the one minting new ones for those who didn't have the chance to exchange it from trader within the 7 days of his visiting.

### **Gelatin & Goblin Bribe**
![Gelatin-400x](https://user-images.githubusercontent.com/104184348/185210350-718be18e-6196-4ddc-a925-911cca7c320b.png)
**Gelatin - Description**
A goblin's favorite dessert.
Slime gels can be cooked in the **Kitchen** to become a **Gelatin**.
Gelatin can be used to bribe the goblins in the map for a listing ticket!
#### Gelatin Requirements:
- 3 slime gel.
- One Gelatin can only be minted once every session/sync(This prevents over minting by people because this is an essential resource).
- Non-Withdrawable
## Phase 4: Slime Mechanics 103 - NFTs and Lootboxes
Also, Gelatin can be used for bribing Goblin Trader to exchange for:
1. WIP Slime House Limited Supply: 10000 NFTs  -> 25 Gelatin and 15 SFL
2. WIPLimited Supply: 5000 NFTs  -> 50 Gelatin and 15 SFL
3. WIP Limited Supply: 500 NFTs  ->  120 Gelatin and 30 SFL
4. WIP Limited Supply: 300 NFT  -> 150 Gelatin and 50 SFL
5.  Loot Box coming from Treasury(WIP Reward Ranges) -> 3 Gelatin Maximum of 3 per day. 15 per week.
Ahhh yes loot boxes one of the things that completes a game and attracts risk takers. As we are in Web3 we definitely want to use those juicy resources in the Community Treasury for this kind of addons / feature. We want to ensure that players are rewarded with what's on the treasury. I do believe that this addon requires a smart contract in the Treasury that will redistribute this resources in the loot box.
## Mutant Slimes(This will be refined to prevent the past case of Mutant Chicken)
Mutant Slimes are the result of slimes producing a new offspring that can help Bumpkins in their Slime Gel production.
Mutation happens after 7 days of minting. With a 1/1000 chance every harvest.
Mutant slimes will be an NFT.
(Old information here is being refined if you already read the old iterations of this proposal)
##Short Summary - (For Programmers)
Phase 1:
- [ ] Add Petri Dish to Mysterious Trader
- [ ] Create Scientist NPC ( 2 UI Tabs: 1 Exchange/Buy 2 Craft)
- [ ] Make Petri Dish as Requirements in Exchange Tab of Scientist NPC
- [ ] Add Slime and Slime Gel
- [ ] Create Functionalities and Behavior of Slime
- [ ] Produce Slime Gels
Phase 2:
- [ ] Add Potions
- [ ] Add Functionalities of Potion
- [ ] Add Potions in 2nd UI Tab of Scientist NPC
- [ ] Add Functionalities of Crafting Potions
- [ ] Add Functionalities of Crafting Petri Dish
**NOTE**: Phases are not final just a temporary lists if this proposal is approved.

### Some Notes to Read if you read until here xD
I know this feature will be loved by fellow Bumpkins, this feature will be a good social experiment with how they will use their slimes, either they will use it for creating Potions for their farm, create Petri Dishes to make it available to other players(non active and new), and in the future, craft Gelatins in the Kitchen for Trading Tickets or HODL Gelatins for Limited NFTs that can help them in their Slime farms, and feel the reward of the Community Treasury.
I do think that this feature can be created cause some of the features mentioned above will only need code reusing besides the randomness feature.

**Additional Note:** 
- Mechanics will change time by time based on core devs feedback
- Arts/Assets will be inserted in the coming days to give more idea on this feature.

(Credits to @JcEii for the art design samples and sfwhat for the SFL fonts)

**Comment by ShiningMinions** [comment]
This is a really solid proposal. Here are the things that really stand out to me as awesome:

1. Slime totally makes sense as needed to create potions, and it's easy to see why slime would have utility in this way. This gives slime a consistent value and use in the game.
2. I absolutely love how you thought of this idea, but more importantly, how you also tied-it-into game lore. The idea of the goblins "loving slime" makes sense. It also makes sense that you can "bribe goblins" using slime to get some perks.
3. What I am huge fan of in your proposal, is how you can use slime to "bribe for lootboxes from the goblin treasury". That is seriously one fantastic idea. To be frankly honest, I can't stand the whole "giveaway culture" of the game at this current moment - it makes the game feel too centralized, attracts low-quality players, and resources are given at the mercy of whoever the core team decides (imagine if your taxes IRL went to people in a lottery-style fashion). Instead, in my humble opinion, I think it would be much more favorable to have code on a smart contract to make this mechanic 100% trustless. (i.e.) I mint slime, I bribe goblins with it, and I get a lootbox from the goblin treasury. Awesome idea worth exploring.

Suggestions to Consider:

The only thing I would have you consider is the "mutant slime" mechanic. This sounds very similar to mutant chickens, which is great - I love mutant chickens! The only thing is that I think if there are going to be "infinite" mutant slimes, there should be a mechanism to burn these mutant slimes (this is also true with mutant chickens). The problem with mutant chickens is that theoretically there is no limit, so the value of these will eventually reach $0. There should be some incentive to burn these to keep supply regulated (like perhaps when food comes out, you can "cook" your mutant chickens/slimes for food + additional perks).

**Reply by eodestroyer** [reply]
Maybe a line of soups could be introduced, requiring beef or chicken broth. ;)

**Comment by JcEii** [comment]
Mutant slime or Hybrid slime? 👀 Supporttttt, yes this can play a big role in potions. It can be used too in kitchen, like gelatin related dishes. (Example of it is coffee jelly or gelatin desset). 

As a moral support pri...(Just an example 😂)
![slimeee](https://user-images.githubusercontent.com/105348036/184340509-4d1017dd-1e98-449c-a3f1-f50b055ba145.gif)

Aseprite (you can use it too but its up to you):
[Slime by jc eii.zip](https://github.com/sunflower-land/sunflower-land/files/9315755/Slime.by.jc.eii.zip)

**Reply by JcEii** [reply]
This one looks good...
![slimeee2](https://user-images.githubusercontent.com/105348036/184364685-964ba8a7-f2a0-4aaf-8ef5-fd54a241709f.gif)

![sample(shadow)](https://user-images.githubusercontent.com/105348036/184480555-8e8d363d-dc4c-44c4-aa22-c0934eb0762d.gif)
(credits to sfwhat's own made sfl font)


--I'm very excited, goodluckkk <3--

Aseprite:
[yellow slime .zip](https://github.com/sunflower-land/sunflower-land/files/9316736/yellow.slime.zip)
[sample.zip](https://github.com/sunflower-land/sunflower-land/files/9331277/sample.zip)

**Reply by LittleEins** [reply]
This are great assets for this bouncy creatures <3 We definitely need more of it <3 and I'll also have to refine/change the mechanics regarding Mutant Slimes that was also mentioned by @ShiningMinions <3

**Comment by esp123** [comment]
Great proposal master eins!!! A pet for Bumpkins, a food for Goblins... I hope we can see slimes in future "adventure" gameplays and lore!!! LFG 🔥🔥🔥

**Comment by adamhannigan** [comment]
Awesome work @LittleEins The idea is incredible!

These are super cute and would be great for engagement. As mentioned above, it feels like a perfect use case for potions.

In terms of game design, we want to bring this down to its most simplest/basic game mechanic. Gelatin, Mutant Slimes + Goblin Exchange are nice add-ons but not core to the concept.

If you had to launch the most slimmed down version of this feature, what would that look like? We need to ensure that the feature by itself is fun without any of the add-ons

**Reply by LittleEins** [reply]
Hi @adamhannigan , Thank you for noticing this bouncy idea <3 Yes it really is a good feature together with potions! 
If we have to launch this feature in its basic form:
1. Mint Slime for 10 SFL.
2. Feed Crops to Slime (Using the scaling measure system and machine logic)
3. Wait for 6 hours to produce Slime Gel(depending on points from scaling measure system)
And that's basically it! The ingredients for the potion will be up to you and slime gels will be the essential ingredient for those potions!

**Reply by adamhannigan** [reply]
That sounds good to me! This will fit in nicely with potions 👌 

One thing we need to be careful of is inflation in this case though. I.e. Farmers mint/grow slimes -> create potions -> grow more crops -> collect more SFL. 

After a few weeks the net effect will be an increase of SFL in circulation, instead of a burn. Do you have any ideas how the burn can exceed the mint?

**Reply by LittleEins** [reply]
I do believe if we add the Bribing feature of gelatin to Goblins in Exchange for a free listing ticket would make sure that slime gels production will also be balanced knowing that people would also want to use IGT(Ingame Trading) but is being limited because of the current prices(0.25$ is small to some countries but also is big to other countries) in this way we also reward active farmers with their grown crops to make sure that they can keep on using IGT.
If slime production is balanced we're sure that players will have to formulate strategies in hoarding crops for slimes or selling it for SFL. We also could do more burns than mint when we came up to more Slime NFTs that can help slime production and engagement(which will come as an addons to further balance it out). 
As I say in the last part of the discussion slimes could be a good social experiment to see what player does with their slime gels be it for market(IGT), for Potions(more crops == more slime gels), for limited NFT(Hodl for something to collect!)

**Reply by LittleEins** [reply]
We're also expecting a future NFT slime who adds the ability to eat woods,stone,iron and gold for a more point allocation to maximize slime production or a different byproduct.

**Reply by LittleEins** [reply]
Let's not forget the point allocation to the feeding scale system proposed which also has limitations(upset stomach) based on the maximum seed supply in shop which also employs more consideration to players on what crops they will use (only 3 slots can be used to allocate points) and also the randomness from the scale also adds a thrill for players whether they risk for a 50% chance on getting the maximum output or maximize the production by allocating long time crops(wheat,radish,parsnip) this would also indirectly affect Mysterious Parsnip where it is overly underrated at the moment.

**Reply by LittleEins** [reply]
@adamhannigan I updated this discussion hoping you can read the new idea I added! I also have the assets files ready.

**Reply by adamhannigan** [reply]
Thanks for the comments!

The tough part of designing upcoming features is to try make them have a net-negative effect on the ecosystem - i.e. they drain more resources than they take.

As game designers, it is hard to design something that overall negatively affects the player but in a Web3 ecosystem this is what we often need to do.

For any 'random' values I would suggest that there are some extremely rare rewards and a few average rewards. When all of the probabilities are combined together, the rewards actually have an average negative. 

Players will engage to try get the rare rewards, but will often be left with less than they put in. This ensures that the feature actually burns more resources than mints them. Do you have some thoughts how we can lessen the rewards but make it appear worthwhile for players?

**Reply by LittleEins** [reply]
I do believe that all those systems working together would create an equilibrium to the ecosystem. It's also inhumanely possible to maximize slime gels output manually by farming unless you buy more crops in IGT. There is a time gating for slimes before they can molt(produce slime gels) which is 6 hours. 
The rewards were designed to be more rewarding through strategies built based on the randomness and limitations of Slimes. It will be the player's decision what they will do with their Slime's help. 
I want to discuss more about it in DM if its okay for you @adamhannigan

**Reply by LittleEins** [reply]
@adamhannigan I updated the discussion again I hope you're patiently reading it with every iterations!

**Comment by S2LKR** [comment]
Just read it for the first time. While I don’t have additional idea to comment on, just wanted to note how comprehensive the proposal is and slime idea seem extremely interesting as well as how very cute they are. Def would be great to have around as a pet!

**Reply by LittleEins** [reply]
I'm glad that you're interested with it xD and I'm extremely sorry for the maths xD

**Comment by Lisarosekisses** [comment]
Great proposal i like it goodluck 😉

**Comment by Life-Is-Great** [comment]
I love this one.  🔥 🔥 🔥 
`Einstein did change the world 😄 `
